tests/rule_based_profiler/parameter_builder/test_simple_date_format_string_parameter_builder.py

```diff
@@ -0,0 +1,156 @@
+    }
+
+    date_format_string_parameter: SimpleDateFormatStringParameterBuilder = (
+        SimpleDateFormatStringParameterBuilder(
+            name="my_simple_date_format_string_parameter_builder",
+        )
+    )
+
+    assert date_format_string_parameter.CANDIDATE_STRINGS == candidate_strings
+    assert date_format_string_parameter._candidate_strings == candidate_strings
+    assert date_format_string_parameter._threshold == 1.0
+
+
+def test_simple_date_format_parameter_builder_zero_batch_id_error():
+    date_format_string_parameter: SimpleDateFormatStringParameterBuilder = (
+        SimpleDateFormatStringParameterBuilder(
+            name="my_simple_date_format_string_parameter_builder",
+        )
+    )
+    parameter_container: ParameterContainer = ParameterContainer(parameter_nodes=None)
+    domain: Domain = Domain(domain_type=MetricDomainTypes.COLUMN)
+
+    with pytest.raises(ge_exceptions.ProfilerExecutionError) as e:
+        date_format_string_parameter._build_parameters(
+            parameter_container=parameter_container, domain=domain
+        )
+
+    assert (
+        str(e.value)
+        == "Utilizing a SimpleDateFormatStringParameterBuilder requires a non-empty list of batch identifiers."
+    )
+
+
+def test_simple_date_format_parameter_builder_alice(
+    alice_columnar_table_single_batch_context,
+):
+    data_context: DataContext = alice_columnar_table_single_batch_context
+
+    batch_request = {
+        "datasource_name": "alice_columnar_table_single_batch_datasource",
+        "data_connector_name": "alice_columnar_table_single_batch_data_connector",
+        "data_asset_name": "alice_columnar_table_single_batch_data_asset",
+    }
+
+    candidate_strings: set[str] = {
+        "%Y-%m-%d",
+        "%Y-%m-%d %H:%M:%S",
+        "%m-%d-%Y",
+        "%Y-%m-%dT%z",
+        "%y-%m-%d",
+    }
+    metric_domain_kwargs = {"column": "event_ts"}
+
+    date_format_string_parameter: SimpleDateFormatStringParameterBuilder = (
+        SimpleDateFormatStringParameterBuilder(
+            name="my_simple_date_format_string_parameter_builder",
+            metric_domain_kwargs=metric_domain_kwargs,
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+    )
+
+    assert date_format_string_parameter.CANDIDATE_STRINGS == candidate_strings
+    assert date_format_string_parameter._candidate_strings == candidate_strings
+    assert date_format_string_parameter._threshold == 1.0
+
+    parameter_container: ParameterContainer = ParameterContainer(parameter_nodes=None)
+    domain: Domain = Domain(
+        domain_type=MetricDomainTypes.COLUMN, domain_kwargs=metric_domain_kwargs
+    )
+
+    assert parameter_container.parameter_nodes is None
+
+    date_format_string_parameter._build_parameters(
+        parameter_container=parameter_container, domain=domain
+    )
+
+    assert len(parameter_container.parameter_nodes) == 1
+    assert parameter_container.parameter_nodes[
```

Comment:
@NathanFarmer Would it make sense to instead test the parameter_container/node by issuing a `get_parameter_value()` statement and give it the parameter name using our supported "$parameter" (or $variable) syntax -- and asserting on the result?  Examples can be found in `tests/rule_based_profiler/test_rule.py` and the pattern looks like this:
```
    assert (
        get_parameter_value_by_fully_qualified_parameter_name(
            fully_qualified_parameter_name=fully_qualified_parameter_name_for_value,
            domain=domain,
            variables=variables_multi_part_name_parameter_container,
            parameters=rule_with_parameters.parameters,
        )
        == value
    )
```

This works only if you have the domain object -- which appears that we do here.  Thanks!

Reply:
Thank you for making me aware of this function! ✅